src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,404 @@
+     * file.
+     */
+    public static final String MSG_JAVADOC_TAG_ORDER = "javadoc.tag.order";
+
+    /** Compiled regexp to match Javadoc tags that take an argument. */
+    private static final Pattern MATCH_JAVADOC_ARG = CommonUtil.createPattern(
+            "^\\s*(?>\\*|\\/\\*\\*)?\\s*@(throws|exception|param)\\s+(\\S+)\\s+\\S*");
+    /** Compiled regexp to match Javadoc tags with argument but with missing description. */
+    private static final Pattern MATCH_JAVADOC_ARG_MISSING_DESCRIPTION =
+        CommonUtil.createPattern("^\\s*(?>\\*|\\/\\*\\*)?\\s*@(throws|exception|param)\\s+"
+            + "(\\S[^*]*)(?:(\\s+|\\*\\/))?");
+
+    @Override
+    public final int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.METHOD_DEF,
+        };
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+
+        final FileContents contents = getFileContents();
+        final int lineNo = ast.getLineNo();
+        final TextBlock textBlock = contents.getJavadocBefore(lineNo);
+
+        if (textBlock != null) {
+            final List<DetailAST> methodParams = getParameters(ast);
+            final List<DetailAST> methodTypeParams = CheckUtil.getTypeParameters(ast);
+            final List<JavadocTag> javadocParams = getMethodTags(textBlock);
+
+            // We will check order separately for methodParams and methodTypeParams
+            final List<JavadocTag> javadocParam =
+                checkMethodTypeParamsOrder(methodTypeParams, javadocParams);
+            checkMethodParamsOrder(methodParams, javadocParam);
+        }
+    }
+
+    /**
+     * Checks order of @param tags for matching method type parameters only.
+     *
+     * @param methodTypeParams list of method type parameters
+     * @param javadocParams list of javadoc parameters
+     * @return list of javadoc parameters
+     */
+    private List<JavadocTag> checkMethodTypeParamsOrder(final List<DetailAST> methodTypeParams,
+                                            final List<JavadocTag> javadocParams) {
+
+        final ListIterator<JavadocTag> javadocTagIt = javadocParams.listIterator();
+        final ListIterator<DetailAST> typeParamsIt = methodTypeParams.listIterator();
+
+        while (typeParamsIt.hasNext() && javadocTagIt.hasNext()) {
+            final JavadocTag tag = javadocTagIt.next();
+            if (!tag.isParamTag()) {
+                continue;
+            }
+
+            javadocTagIt.remove();
+
+            final DetailAST typeParam = typeParamsIt.next();
+            final String arg1 = tag.getFirstArg();
+            if (CommonUtil.startsWithChar(arg1, '<') && CommonUtil.endsWithChar(arg1, '>')) {
+                if (!typeParam.findFirstToken(TokenTypes.IDENT).getText()
```

Comment:
`[ERROR] src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java:[284,13] (extension) MoveVariableInsideIf: Variable 'typeParam' can be moved inside the block at line '286' to restrict runtime creation.`
whenever i move this line i start getting check logic errors(extra violations), that's strange but it's happening.